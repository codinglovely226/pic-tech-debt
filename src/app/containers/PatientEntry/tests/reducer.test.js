/* eslint-disable */

import PatientReducer from '../reducer';

import {
  replacePatientMedicalRecordNumber,
  replacePatientName,
  updatePatientAge,
  updatePatientAgeGroup,
  updatePatientAgeYears,
  updatePatientAnimalType,
  updatePatientCategory,
  updatePatientCircumstance,
  updatePatientLocation,
  updatePatientMedicalRecordNumber,
  updatePatientName,
  updatePatientOccupation,
  updatePatientPoisonSeverityScore,
  updatePatientRiskAssessment,
  updatePatientSex,
  updatePatientWeight,
} from '../actions/case';

describe('PatientReducer', () => {
  const appKey = { id: 'appKeyId', key: 'appKey' };
  it('returns the initial state', () => {
    expect(PatientReducer(undefined, {})).toMatchSnapshot();
  });
  it('update patient age', () => {
    expect(PatientReducer({}, replacePatientMedicalRecordNumber(1, 'mrn', appKey))).toMatchSnapshot();
  });
  it('update patient age', () => {
    expect(PatientReducer({}, replacePatientName(1, 'Patient Name', appKey))).toMatchSnapshot();
  });
  it('update patient age', () => {
    expect(PatientReducer({}, updatePatientAge(1, 10, 1, 'months', 'years'))).toMatchSnapshot();
  });
  it('update patient age group', () => {
    expect(PatientReducer({}, updatePatientAgeGroup(1, 'months', 'years'))).toMatchSnapshot();
  });
  it('update patient age years', () => {
    expect(PatientReducer({}, updatePatientAgeYears(1, '', '1.5'))).toMatchSnapshot();
  });
  it('update patient animal type', () => {
    expect(PatientReducer({}, updatePatientAnimalType(1, 'dog', 'bird'))).toMatchSnapshot();
  });
  it('clear patient animal type if human category', () => {
    expect(PatientReducer({'category': 'bird'}, updatePatientCategory(1, 'animal', 'human'))).toMatchSnapshot();
  });
  it('update patient category', () => {
    expect(PatientReducer({}, updatePatientCategory(1, 'human', 'animal'))).toMatchSnapshot();
  });
  it('update patient circumstance', () => {
    expect(PatientReducer({}, updatePatientCircumstance(1, 'Old circumstance', 'New circumstance'))).toMatchSnapshot();
  });
  it('update patient location', () => {
    expect(PatientReducer({}, updatePatientLocation(1, 'Work', 'Hospital'))).toMatchSnapshot();
  });
  it('update patient MRN', () => {
    expect(PatientReducer({}, updatePatientMedicalRecordNumber(1, 5678, 4321, appKey))).toMatchSnapshot();
  });
  it('update patient name', () => {
    expect(PatientReducer({}, updatePatientName(1, 'Joe Blogs', 'John Smith', appKey))).toMatchSnapshot();
  });
  it('update patient occupation', () => {
    expect(PatientReducer({}, updatePatientOccupation(1, 'Student', 'Salesperson'))).toMatchSnapshot();
  });
  it('update patient poison severity score', () => {
    expect(PatientReducer({}, updatePatientPoisonSeverityScore(1, 3, 4))).toMatchSnapshot();
  });
  it('update patient risk assessment', () => {
    expect(PatientReducer({}, updatePatientRiskAssessment(1, 'Old assessment', 'New assessment'))).toMatchSnapshot();
  });
  it('update patient sex', () => {
    expect(PatientReducer({}, updatePatientSex(1, 'female', 'male'))).toMatchSnapshot();
  });
  it('update patient weight', () => {
    expect(PatientReducer({}, updatePatientWeight(1, 65, 68, 'kg', 'kg'))).toMatchSnapshot();
  });
});
