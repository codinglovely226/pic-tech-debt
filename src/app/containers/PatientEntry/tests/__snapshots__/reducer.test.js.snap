exports[`PatientReducer clear patient animal type if human category 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer returns the initial state 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient MRN 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": 4321,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient age 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "encryptedString",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient age 2`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "encryptedString",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient age 3`] = `
Object {
  "age": 1,
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient age group 1`] = `
Object {
  "age": "",
  "ageGroup": "years",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient age years 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "1.5",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient animal type 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "bird",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient category 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "animal",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient circumstance 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "New circumstance",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient location 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "Hospital",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient name 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "John Smith",
  "nameEnc": "encryptedString",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient occupation 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "Salesperson",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient poison severity score 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": 4,
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient risk assessment 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "New assessment",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient sex 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "male",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": "",
  "weightUnits": "kg",
}
`;

exports[`PatientReducer update patient weight 1`] = `
Object {
  "age": "",
  "ageGroup": "adult",
  "ageUnits": "years",
  "ageYears": "",
  "animalType": "",
  "category": "human",
  "circumstance": "",
  "gender": "",
  "locality": Object {
    "address1": "",
    "address2": "",
    "city": "",
    "country": "",
    "geometry": Object {},
    "isoCountry": "",
    "location": "",
    "name": "",
    "placeId": "",
    "postcode": "",
    "state": "",
    "telephone": "",
  },
  "medicalRecordNumber": null,
  "medicalRecordNumberEnc": "",
  "name": "",
  "nameEnc": "",
  "occupation": "unknown",
  "poisonSeverityScore": "Moderate",
  "pregnant": Object {
    "lactating": "unknown",
    "status": "unknown",
    "trimester": "unknown",
    "trimesterUnknown": null,
  },
  "riskAssessment": "Moderate Risk",
  "weight": 68,
  "weightUnits": "kg",
}
`;
